### DISCUSSION TITLE
I'm getting a 404 on any page requests through mfscgiserv

### DISCUSSION THREAD

**Discussion by vineet-endpoint** [moosefs/moosefs#547] [General]
I'm getting a 404 on any page requests through mfscgiserv. See below. Any idea how I can debug this?
I started a second mfsmaster called mfsmaster2 so that I can copy the files over. I've setup mfsfaster and mfsmaster2 in /etc/hosts. 

root@host70:~# mfscgiserv -fv
lockfile created and locked
starting simple cgi server (host: any , port: 9425 , rootpath: /usr/share/mfscgi)
Asynchronous HTTP server running on port 9425
Vineets-MBP - - [29/Aug/2023 10:03:05] "GET / HTTP/1.1" 404
Vineets-MBP - - [29/Aug/2023 10:04:04] "GET /index.html HTTP/1.1" 404

**Comment by chogata** [comment]
Did you check the content of your `/usr/share/mfscgi` directory? Are the required files there, are the access settings to the directory and files correct?

Did you install from sources or packages? Remember, that cgiserv and cgi are 2 different packages, if you installed only the cgiserv, you won't have the CGI scripts installed.

**Comment by vineet-endpoint** [comment]
I'm on ubuntu 22.04, I don't have the folder /usr/share/mfscgi. Do I need to copy the files from another folder. I removed moosefs-cgiserv and moosefs-cgi and reinstalled them.  I grep for cgi on the root folder and it seems the files are in /usr/share/moosefs-cgi and /usr/share/mfscgi. 

I ran it with -R /usr/share/moosefs-cgi and now it works but it is going to server mfsmaster and mfsmaster2. How do I make the cgi connect to the master running on the same host.

**Comment by chogata** [comment]
You can add the `masterhost` paremeter with a hostname value in your url to connect to any master. Like this:

`http://your.cgi.host:9425/mfs.cgi?masterhost=your.master.host`

**Comment by vineet-endpoint** [comment]
Thank you!
